hw3/got-stats/src/pages/Search/Search.jsx

```diff
@@ -0,0 +1,88 @@
+import React, { useEffect, useState } from 'react';
+import Dropdown from 'react-dropdown';
+import 'react-dropdown/style.css';
+
+const url = 'https://thronesapi.com/api/v2/Characters';
+
+const Search = () => {
+  const [characters, setCharacters] = useState([]);
+  const [searchResult, setSearchResult] = useState([]);
+  const [searchString, setSearchString] = useState('');
+  const [showResultToggle, setShowResultToggle] = useState(false);
+
+  useEffect(() => {
+    const fetchApiData = async () => {
+      try {
+        const response = await fetch(url);
+        const data = await response.json();
+        setCharacters(data);
+      } catch (error) {
+        console.error('An error occurred:', error);
+      }
+    };
+    fetchApiData();
+  }, []);
+
+  const handleSearch = () => {
+    const results = characters.filter((character) =>
+      character.fullName
+        .trim()
+        .toLowerCase()
+        .includes(searchString.toLowerCase())
```

Comment:
Search could use more refining. Should only return if input matches exactly.  I get a results when entering in just 1 letter.  
For example: 
![image](https://github.com/sarvesh2910/CS564-frontend-homework/assets/97830813/be7f72c3-32a8-4047-bb1c-6806b96ceb7b)

Reply:
updated.

Reply:
Tested and works now.